Tests/Test_SDL.cpp

```diff
@@ -0,0 +1,228 @@
+ * published by the Free Software Foundation, either version 3 of the         *
+ * License, or (at your option) any later version.                            *
+ *                                                                            *
+ * dirtsand is distributed in the hope that it will be useful,                *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the              *
+ * GNU Affero General Public License for more details.                        *
+ *                                                                            *
+ * You should have received a copy of the GNU Affero General Public License   *
+ * along with dirtsand.  If not, see <http://www.gnu.org/licenses/>.          *
+ ******************************************************************************/
+
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+
+#include <catch2/catch.hpp>
+#include <string_theory/format>
+
+#include <unistd.h>
+
+#include "SDL/DescriptorDb.h"
+#include "SDL/StateInfo.h"
+
+static const ST::string s_SdlDescriptor = ST_LITERAL(R"(
+    STATEDESC Test
+    {
+        VERSION 1
+
+        VAR BOOL    bTestVar1[1]    DEFAULT=0
+        VAR BOOL    bTestVar2[1]    DEFAULT=1    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar3[1]    DEFAULT=0    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar4[1]    DEFAULT=100
+    }
+
+    STATEDESC Test
+    {
+        VERSION 2
+
+        VAR BOOL    bTestVar1[1]    DEFAULT=0
+        VAR BOOL    bTestVar2[1]    DEFAULT=1    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar3[1]    DEFAULT=0    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar4[1]    DEFAULT=100
+
+        VAR BYTE    bTestVar5[1]    DEFAULT=50
+    }
+
+    STATEDESC Barney
+    {
+        VERSION 1
+    }
+)");
+
+static bool LoadDescriptors()
+{
+    static bool result = false;
+    if (result)
+        return true;
+
+    ST::string sdlFilePath;
+    char tempDir[256]{};
+    const char kTemplate[] = "DirtSandSDLTestXXXXXX";
+    strcpy(tempDir, kTemplate);
+    do {
+        char* tempDirResult = mkdtemp(tempDir);
+        if (!tempDirResult) {
+            fprintf(stderr, "Failed to create a temporary directory for SDL.\n");
+            break;
+        }
+
+        sdlFilePath = ST::format("{}/Test.sdl", tempDirResult);
+        fprintf(stderr, "%s\n%s\n%s\n", sdlFilePath.c_str(), tempDir, tempDirResult);
+        FILE* f = fopen(sdlFilePath.c_str(), "w+");
+        if (!f) {
+            fprintf(stderr, "Failed to create a file in our temporary SDL directory.\n");
+            break;
+        }
+        fwrite(s_SdlDescriptor.c_str(), sizeof(char), s_SdlDescriptor.size(), f);
+        fclose(f);
+
+        SDL::DescriptorDb::LoadDescriptors(tempDirResult);
+        result = true;
+    } while (0);
+
+    if (!sdlFilePath.empty())
+        unlink(sdlFilePath.c_str());
+    if (*tempDir)
```

Comment:
This test should be against `tempDirResult` -- `mkdtemp` doesn't clear the template upon failure, as far as I can tell.